### PR TITLE
Add: UserEvent and EventRole Services 

### PR DESCRIPTION
**Pull request by Mukhammadzokir**
#17,
#16

### CODE REVIEW

File: src/MDC.Server.Service/Services/EventRoleService.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        var data = await this._eventRoleRepository
+            .SelectAll()
+            .Where(e => e.Id == id)
+            .FirstOrDefaultAsync();
+        if(data is null)
+            throw new MDCException(404,"EventRole is not found");
+
+        var mappedData = this._mapper.Map(dto, data);
+        mappedData.UpdatedAt = DateTime.UtcNow;
+        await this._eventRoleRepository.UpdateAsync(mappedData);
```
Comment (Nozimjon04) [inline src/MDC.Server.Service/Services/EventRoleService.cs:53]: This is extra work as you did not use the asnotracking

File: src/MDC.Server.Service/Services/UserEventService.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    public async Task<UserEventForResultDto> AddAsync(UserEventForCreationDto dto)
+    {
+        var data = this._userEventRepository
+            .SelectAll()
+            .Where(u => u.UserId == dto.UserId && u.EventId == dto.EventId && u.RoleId == dto.RoleId)
+            .AsNoTracking()
+            .FirstOrDefaultAsync();
+        if(data is not null)
+            throw new MDCException(409, "UserEvent is already exist");
+
```
Comment (Nozimjon04) [inline src/MDC.Server.Service/Services/UserEventService.cs:35]: Check for User and Event exisstence

File: src/MDC.Server.Service/Services/UserEventService.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    public async Task<UserEventForResultDto> ModifyAsync(long id, UserEventForUpdateDto dto)
+    {
+        var data = await this._userEventRepository
+            .SelectAll()
+            .Where(u => u.Id == id)
+            .FirstOrDefaultAsync();
+        if(data is null)
+            throw new MDCException(404,"UserEvent is not found");
+
```
Comment (Nozimjon04) [inline src/MDC.Server.Service/Services/UserEventService.cs:50]: same problem with added method

File: src/MDC.Server.Service/Services/UserEventService.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        var data = await this._userEventRepository
+            .SelectAll()
+            .Include(u => u.User)
+            .Include(u => u.Event)
+            .Include(u => u.Role)
+            .AsNoTracking()
+            .ToPagedList(@params)
+            .ToListAsync();
+        if(data is null)
+            throw new MDCException(404,"Data are not found");
```
Comment (Nozimjon04) [inline src/MDC.Server.Service/Services/UserEventService.cs:80]: No need for checking here!

**Comment by Mukhammadzokir** [comment]
update some

**Pull request closed without merging**